### PR TITLE
Add a new input to Get Connections that allows you to choose between System Pins, DUT Pins, or All Pins

### PR DESCRIPTION
**Pull request by pbirkhol-ni**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/measurement-services-labview/blob/main/CONTRIBUTING.md).

### What does this Pull Request accomplish?

[AB#2543570](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2543570)

This PR adds a new input to the Get Connections VIs: all 3 VIMs, the Impl VI, and all of the driver implementations. The new typedef allows you to filter pins based on the pin type: `DUT and System Pins` or `System Pins`

I also had to update any tests that used the VIMs, which was a lot of them. The new input is required for the VIMs, so every test had to be updated to add the required input. No other changes were made to tests.

### What testing has been done?

All auto tests pass. I did not write any new auto tests, although we should test the new input. The reason I didn't write new tests yet is because we need to rethink error handling with this new input. When I address that as part of [AB#2543570](https://dev.azure.com/ni/94b22d7b-ad7b-4f5e-88f0-867910f91c94/_workitems/edit/2543570), I will write the new tests.

### CODE REVIEW

File: Source/Runtime/MeasurementLink Session Management/Session Reservation/Get Connections Impl.vi
Comment (pbirkhol-ni) [inline Source/Runtime/MeasurementLink Session Management/Session Reservation/Get Connections Impl.vi:1]: The red circle is the only new logic in this PR. Every other change is just changing the connector pane of VIs to add the new input and then wiring it. I also modified documentation for each VI.
image

**Comment by dixonjoel** [comment]
@pbirkhol-ni I guess we're still discussing and deciding on this? Or should I go ahead and review it?

**Comment by pbirkhol-ni** [comment]
> @pbirkhol-ni I guess we're still discussing and deciding on this? Or should I go ahead and review it?

You can hold off for now. I'll let you know when it is ready for review.

**Comment by pbirkhol-ni** [comment]
> > @pbirkhol-ni I guess we're still discussing and deciding on this? Or should I go ahead and review it?
> 
> You can hold off for now. I'll let you know when it is ready for review.

This is ready to review now. The only change that I made is that the new enum now only has two options: `DUT and System Pins` and `System Pins`. It previously also allowed `DUT Pins`, which is a feature that Python doesn't support. We can add that feature later if we want.

**Review by dixonjoel — APPROVED** [review]

**Pull request merged**
